### DISCUSSION TITLE
CLI and/or CI/CD for Tabular Editor 3?

### DISCUSSION THREAD

**Discussion by ArtBelch** [TabularEditor/TabularEditor3#675] [🙋♀️TE3 Q&A] [answered]
Hi,

Is there a CLI for tabular editor 3 so we can handle CI/CD as we can for TE2? What are the future plans for handling CI/CD in general?

Nicklas

**Comment by otykier** [comment, marked as answer]
TE2 CLI is fully compatible with the TE3 file format. We are, however, planning a standalone CLI tool with some additional capabilities.

**Comment by 2bfr33** [comment]
Hi @otykier, 
has TE3 support for headless / CI-CD deployment via XMLA already been implemented, or is it still planned? We are looking to deploy TMDL-based models to Power BI / Fabric workspaces via XMLA endpoints in CI/CD pipelines. And so far deploy with TE2 is not working for TMDLs. thanks.

**Reply by mlonsk** [reply]
Hi @2bfr33 
Tabular Editor 3 does not have a CLI for headless deployment yet. We are actively investigating our options in this regard, but I don't have a timeline I can share. 
To my knowledge TE2 should work with TMDL. Which version are you using of it?

**Reply by 2bfr33** [reply]
Hi @mlonsk
thanks for reply. latest version but cicd pipeline was old and I have refactor it and can confirm that it is working now with tmdl! thanks!

**Reply by mlonsk** [reply]
Hi @2bfr33 
Good to hear, glad you got it working again. 
Best
Morten
